issue99_test.go

```diff
@@ -0,0 +1,78 @@
+package zmq4
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+)
+
+var (
+	wg     sync.WaitGroup
+	outMsg Msg
+	inMsg  Msg
+)
+
+func requester(t *testing.T) {
+
+	req := NewReq(context.Background())
+	defer req.Close()
+
+	err := req.Dial("tcp://localhost:5559")
+	if err != nil {
+		wg.Done()
+		t.Fatalf("could not dial: %v", err)
+	}
+
+	// Test message w/ 3 frames
+	outMsg = NewMsgFromString([]string{"ZERO", "Hello!", "World!"})
+	err = req.Send(outMsg)
+	if err != nil {
+		wg.Done()
+		t.Fatalf("failed to send: %v", err)
+	}
+
+	inMsg, err = req.Recv()
+	wg.Done()
+	if err != nil {
+		t.Fatalf("failed to recv: %v", err)
+	}
+}
+
+func responder(t *testing.T) {
+
+	//  Socket to talk to clients
+	rep := NewRep(context.Background())
+	defer rep.Close()
+
+	err := rep.Listen("tcp://*:5559")
+	if err != nil {
+		wg.Done()
```

Comment:
ditto.

please add a single `defer wg.Done()` at the top of that function, and remove all other occurrences of `wg.Done()` below.